### PR TITLE
Guicing more

### PR DESCRIPTION
**Pull request by eishay**

### CODE REVIEW

File: shoebox/app/com/keepit/controllers/ext/ExtEventController.scala [hunk 1]
```diff
@@ -37,7 +38,9 @@ class ExtEventController @Inject() (
   db: Database,
   userExperimentRepo: UserExperimentRepo,
   userRepo: UserRepo,
-  persistEventPlugin: PersistEventPlugin)
+  persistEventPlugin: PersistEventPlugin,
+  implicit private val clock: Clock,
```
Comment (gmethvin) [inline shoebox/app/com/keepit/controllers/ext/ExtEventController.scala:42]: Do you need the `private val`? You'll still have access to the constructor parameter in the class.

File: shoebox/test/com/keepit/search/MainSearcherTest.scala [hunk 2]
```diff
@@ -52,7 +54,9 @@ class MainSearcherTest extends Specification with DbRepos {
         resultClickTracker,
         browsingHistoryTracker,
         clickHistoryTracker,
-        inject[FakePersistEventPluginImpl])
+        inject[FakePersistEventPluginImpl],
+        clock,
```
Comment (gmethvin) [inline shoebox/test/com/keepit/search/MainSearcherTest.scala:58]: You can just do `inject[Clock]` since you're explicitly passing the parameter.

**Comment by gmethvin** [comment]
lgtm

File: shoebox/app/com/keepit/controllers/ext/ExtEventController.scala (hunk 1 above)
Comment (eishay) [inline shoebox/app/com/keepit/controllers/ext/ExtEventController.scala:42]: I want the clock to be passed implicitly and I can't have implicit without declaring it as a val.
I'm not sure weather its better then just passing the clock explicitly around, its arguable.

File: shoebox/test/com/keepit/search/MainSearcherTest.scala (hunk 2 above)
Comment (eishay) [inline shoebox/test/com/keepit/search/MainSearcherTest.scala:58]: right

File: shoebox/app/com/keepit/controllers/ext/ExtEventController.scala (hunk 1 above)
Comment (gmethvin) [inline shoebox/app/com/keepit/controllers/ext/ExtEventController.scala:42]: Okay, didn't realize you needed implicits to be a val. Okay.

**Pull request merged**